player.cpp

```diff
@@ -82,7 +82,7 @@ void player::decelerate() noexcept
     }
     else if(m_player_speed < 0.0)
     {
-        if(abs(m_player_speed) >= m_player_deceleration_backward)
+        if(std::abs(m_player_speed) >= m_player_deceleration_backward)
```

Comment:
abs() is designed for int while std::abs() can deal with float or double

Reply:
https://stackoverflow.com/questions/21392627/abs-vs-stdabs-what-does-the-reference-say
link that might help